spec/domain/people/duplicate_checker_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'spec_helper'
+
+describe People::DuplicateChecker do
+
+  let(:checker) { described_class.new }
+  subject { checker.check }
+
+  context 'check' do
+    it 'creates one duplicate when all attributes match once' do
+      create_duplicate(:top_leader)
+      expect { subject }.to change { PersonDuplicate.count }.by(1)
+    end
+
+    it 'creates two duplicate when all attributes match twice' do
+      create_duplicate(:top_leader)
+      create_duplicate(:top_leader)
+      expect { subject }.to change { PersonDuplicate.count }.by(2)
+    end
+
+    it 'creates one duplicates when names, company name, birthday match and zip code is empty' do
```

Comment:
creates duplicate entry when ....

creates duplicate entry for matching name, company name, birthday but blank zip code